idalib/src/strings.rs

```diff
@@ -0,0 +1,54 @@
+use std::marker::PhantomData;
+
+use crate::ffi::strings::{
+    build_strlist, clear_strlist, get_strlist_qty, idalib_get_strlist_item_addr,
+    idalib_get_strlist_item_length,
+};
+use crate::ffi::BADADDR;
+
+use crate::idb::IDB;
+use crate::Address;
+
+pub struct StringList<'a> {
+    _marker: PhantomData<&'a IDB>,
+}
+
+impl<'a> StringList<'a> {
+    pub(crate) fn new(_: &'a IDB) -> Self {
+        Self {
+            _marker: PhantomData,
+        }
+    }
+
+    // Note: Seems pretty much useless
+    pub fn build(&self) {
```

Comment:
If these aren't needed, then maybe we can just not expose them?

Reply:
I could remove them if you want. From my tests they don't do anything, but IDA SDK's documentation [here](https://cpp.docs.hex-rays.com/strlist_8hpp.html) seems to imply that they're useful, so I kept them just in case I had missed something during testing:
```
While the kernel keeps the string list, it does not update it. The string list is 
not used by the kernel because keeping it up-to-date would slow down IDA 
without any benefit. If the string list is not cleared using clear_strlist(), the 
list will be saved to the database and restored on the next startup.

The users of this list should call build_strlist() if they need an up-to-date 
version.
```
Let me know if I should remove them for the time being.

Reply:
Maybe we can expose `build` as `rebuild` since that's what this description seems to imply is going on? And keep both it and `clear`?

Reply:
Done!